daphne/src/roles.rs

```diff
@@ -55,18 +55,29 @@ where
     fn get_global_config(&self) -> &DapGlobalConfig;
 
     /// Look up the DAP task configuration for the given task ID.
-    async fn get_task_config_considering_taskprov(
+    ///
+    /// If a `report` has been provided, then look for the draft-wang-ppm-dap-taskprov-<nn> extension
+    /// in the report.  If a taskprov task configuration is successfully read from the report,
+    /// the `opt_in_callback` will be called, and if it returns Ok(true) the server will opt-in to the task.
+    /// if it returns Ok(false) or an error then the server will opt-out or return an appropriate error.
+    async fn get_task_config_considering_taskprov<F>(
         &'srv self,
+        version: DapVersion,
         task_id: Cow<'req, Id>,
         report: Option<&ReportMetadata>,
-    ) -> Result<Option<Self::WrappedDapTaskConfig>, DapError>;
+        opt_in_callback: F,
+    ) -> Result<Option<Self::WrappedDapTaskConfig>, DapError>
+    where
+        F: FnOnce(&DapTaskConfig) -> Result<bool, DapError>;
 
     /// Look up the DAP task configuration for the given task ID.
     async fn get_task_config_for(
         &'srv self,
         task_id: Cow<'req, Id>,
     ) -> Result<Option<Self::WrappedDapTaskConfig>, DapError> {
-        self.get_task_config_considering_taskprov(task_id, None)
+        // We use DapVersion::Unknown here as we don't know it and we don't need to
+        // know it as we will not be doing any taskprov task creation.
+        self.get_task_config_considering_taskprov(DapVersion::Unknown, task_id, None, |_| Ok(false))
```

Comment:
In fact, I think the Helper should always know which version was indicated, via the version field of the `DapRequest`. I suppose the Leader would also know based on its version preference (typically the latest draft).

Reply:
The Helper and the Leader do pass the API entry point version used in the cases where taskprov might create a task config.  The "Unknown" version is used in cases where we don't have a report (and thus won't be creating any tasks) and are instead either going to find an existing task (taskprov or not) or find nothing and complain.   The unknown and the false callback are just to adapt the the API so we don't duplicate code.